openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -169,43 +194,148 @@ def run(self, problem):
         self.quantities += list(econs)
 
         for name in self.get_constraints(ctype='eq'):
-            size = con_meta[name]['size']
-            lower = upper = con_meta[name]['equals']
+            meta = con_meta[name]
+            size = meta['size']
+            lower = upper = meta['equals']
 
             # Sparsify Jacobian via relevance
-            wrt = rel.relevant[name].intersection(indep_list)
+            rels = rel.relevant[name]
+            wrt = rels.intersection(indep_list)
             self.sparsity[name] = wrt
 
-            if con_meta[name]['linear'] is True:
+            if meta['linear'] is True:
                 opt_prob.addConGroup(name, size, lower=lower, upper=upper,
                                      linear=True, wrt=wrt,
                                      jac=self.lin_jacs[name])
             else:
+
+                jac = None
+
+                # Additional sparsity for index connections
+                for param in wrt:
+                    sub_conns = sub_param_conns.get(param)
+                    if not sub_conns:
+                        continue
+
+                    rel_idx = set()
+                    for target, idx in iteritems(sub_conns):
+
+                        # If a target of the indexed desvar connection is
+                        # in the relevant path for this constraint, then
+                        # those indices are relevant.
+                        if target in rels:
+                            rel_idx = rel_idx.union(idx)
```

Comment:
instead of doing a union here, I would just do  rel_idx.update(idx), since you just want it to update an existing set, not create a new set and assign it back to rel_idx.